README.md

```diff
@@ -8,21 +8,222 @@
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the Hubspot Automation action API connector in Ballerina, you must have a HubSpot developer account.
+
+ ### Step 1: Create a Hubspot Developer Account
+
+Visit the Hubspot portal (https://developers.hubspot.com/get-started) and create a Developer Account.
+
+
+ ### Step 2: Create a Hubspot Developer Test Account
+
+Visit (https://developers.hubspot.com/beta-docs/getting-started/account-types#developer-test-accounts)and create a hubspot developer test account
+
+ ### Step 3: Create a Hubspot Public App
+
+In your developer account, navigate to the "Apps" section.
+
+Click on "Create App" and provide the necessary details, including the app name and description.
+
+
+ ### Step 4: Initiate the OAuth Flow
+
+Move to the auth tab in the created app and set the permissions there .
+
+Under the OAuth tab you can find the following details ,
+
+
+client_id: Your app's Client ID.
+
+redirect_uri: The URL users will be redirected to after granting access.
+
+scope: A space-separated list of scopes your app is requesting.
+
+![Auth Tab example](docs/setup/resources/auth.png)
+
+
+### Step   5: Add the redirect URL
+Add your redirect url under the redirect urls.
+
+
+![alt text](docs/setup/resources/redirect_url.png)
+
+
+### Step   6: Add the Required Scopes
+
+Go to the relevant API documentation (https://developers.hubspot.com/docs/reference/api/automation/custom-workflow-actions) and find out the required scopes.
+
+
+![Requied Scope doc](docs/setup/resources/scope_doc.png)
+
+
+
+Now come back to the Auth Page and add the required scopes under the Auth tab .
+
+
+
+![Required Scopes](docs/setup/resources/scopes.png)
+
+Now save the app 
+
+![Save the app](docs/setup/resources/save.png)
+
+### Step   7: Obtain the authorization code 
+
+Copy the App installation url and paste it in the web browser.
+
+![Redirect URL ](docs/setup/resources/redirect.png)
 
-## Quickstart
 
-[//]: # (TODO: Add a quickstart guide to demonstrate a basic functionality of the module, including sample code snippets.)
+
+ It wll prompt you to install  the App and then select your deveper test account.
+
+
+
+After selcting the developertest account, you will receive a authorization code displalyed in the browser.
+
+![Obtain the authorization code](docs/setup/resources/authorization_code.png)
+
+### Step   8: Obtain the access token
+
+
+Place your authorization code, client_id and client_secret in the folowing comand and execute it in the terminal
+
+'curl --request POST \
+  --url https://api.hubapi.com/oauth/v1/token \
+  --header 'content-type: application/x-www-form-urlencoded' \
+  --data 'grant_type=authorization_code&code=<code>&redirect_uri=http://localhost:9090&client_id=<client_id>&client_secret=<client_secret>'
+
+If the command executes successfully , you will receive the access token from the response.
```

Comment:
To receive the token, you need a listener to run. Let's add that information here. We don't need the Ballerina code snippet here. Just mention that they need to run a listener locally to listen on port `9090` to receive the token.